lighthouse-core/lib/median-run.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/** @param {LH.Result} lhr @param {string} auditName */
+const getNumericValue = (lhr, auditName) =>
+  (lhr.audits[auditName] && lhr.audits[auditName].numericValue) || NaN;
+
+/**
+ * @param {LH.Result} lhr
+ * @param {number} medianFcp
+ * @param {number} medianInteractive
+ */
+function getMedianSortValue(lhr, medianFcp, medianInteractive) {
+  const distanceFcp = medianFcp - getNumericValue(lhr, 'first-contentful-paint');
+  const distanceInteractive =
+    medianInteractive - getNumericValue(lhr, 'interactive');
+
+  return distanceFcp * distanceFcp + distanceInteractive * distanceInteractive;
+}
+
+/**
+ * @param {Array<LH.Result>} runs
+ * @return {LH.Result|undefined}
+ */
+function computeMedianRun(runs) {
+  const validRuns = runs
+    .filter(run => Number.isFinite(getNumericValue(run, 'first-contentful-paint')))
```

Comment:
i feel a little weird this is silent. i know we don't like console logs but it feels odd we'd silently discard LHRs.

Reply:
Would you prefer to throw instead? I would really rather not `console.log` in something that is supposed to be a library for other projects.

Reply:
Let's throw then and expose the filtering method separately. That lets the caller decide how they want to handle it while exposing as much information as possible.

Reply:
throwing is ... a lot.. but yeah seems good for now. 

speaking of callers, how do you plan on handling it in lhci? :)

Reply:
well we won't be able to use this for another major breaking change since we've already added audits that change scoring, haha

in practice I don't expect it to matter because we rerun the collect up to 3 times if major things failed. if they're doing their own collection and manually uploading though we would silently drop invalid runs. by the time this gets run on the server for LHR selection there's nothing that can really be done about it anyway, that build is gone and no human to really notify other than maybe a warning in some build logs no one looks at.